src/pybind/mgr/cephadm/module.py

```diff
@@ -1040,10 +1043,26 @@ def _check_hosts(self):
             }
         self.set_health_checks(self.health_checks)
 
-    def _check_host(self, host):
+    def _check_host_ping(self, host) -> Optional[str]:
```

Comment:
I'm not sure about using ping.. there are some network environments where ping is not allowed but other things are.

Also, it seems like our *regular* 'check-host' command should time out at 60s too.  Better to just implement that timeout across the board?  That way we don't have an awkward window where we don't realize teh host is down, we do the check-host, and wait 300s or whatever for the longer ssh timeout to happen.

Reply:
Maybe we can add `ConnectTimeout=30` to our ssh config?

Reply:
Yes! :+1: